cmd/utils.go

```diff
@@ -229,3 +231,44 @@ func toS3ETag(etag string) string {
 
 	return etag
 }
+
+// isNetworkOrHostDown - if there was a network error or if the host is down.
+func isNetworkOrHostDown(err error) bool {
+	if err == nil {
+		return false
+	}
+	switch err.(type) {
+	case net.Error:
+		switch err.(type) {
+		case *net.DNSError, *net.OpError, net.UnknownNetworkError:
+			return true
+		case *url.Error:
+			// For a URL error, where it replies back "connection closed"
+			if strings.Contains(err.Error(), "Connection closed by foreign host") {
+				return true
+			}
+		default:
+			if strings.Contains(err.Error(), "net/http: TLS handshake timeout") {
+				// If error is - tlsHandshakeTimeoutError,.
+				return true
+			} else if strings.Contains(err.Error(), "i/o timeout") {
+				// If error is - tcp timeoutError.
+				return true
+			} else if strings.Contains(err.Error(), "connection timed out") {
+				// If err is a net.Dial timeout.
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// GetLastAccess gets last accessed date and time of the object.
+func GetLastAccess(int interface{}) (aTime time.Time) {
+	stat, ok := int.(*syscall.Stat_t)
```

Comment:
We should make this multi platform, this cannot compile properly and work on windows.  Perhaps create a package like this https://github.com/djherbis/atime

Reply:
Was this resolved? I tried to enable caching on Windows and got the error: `ERROR Unable to initialize disk caching: Atime support required for disk caching`. Last access time is enabled on the drive:

```
D:\>fsutil behavior query DisableLastAccess
DisableLastAccess = 0
```

Reply:
@ngbrown caching is only supported on Linux currently, use it accordingly.

Reply:
@harshavardhana Ok. I'm forced to have the primary data drive be a high capacity file share from a network appliance.  I wanted to cache on the local, fast flash drive to reduce network traffic.  If I just have to make an `atime` compatible package for Windows, I can look into that.

Reply:
By all means, all you have to fix is time package which is use to support windows.